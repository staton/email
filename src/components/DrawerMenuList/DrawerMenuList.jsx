import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { MdInbox } from 'react-icons/md';
import DrawerMenuListItem from '../DrawerMenuListItem/DrawerMenuListItem';
import STRINGS from '../../resources/strings';

const propTypes = {
};

const defaultProps = {
};

export class DrawerMenuList extends Component {

	render() {
		return (
            <ul className="DrawerMenuList">
                <DrawerMenuListItem 
                    badge={205}
                    linkTo="/inbox"
                    icon={<MdInbox />}
                    primaryContent={STRINGS.InboxMenu}
                    secondaryContent="test"
                />
                <DrawerMenuListItem 
                    linkTo="/drafts"
                    primaryContent={STRINGS.DraftsMenu}
                />
                <DrawerMenuListItem 
                    linkTo="/sent"
                    primaryContent={STRINGS.SentMenu}
                />
                <DrawerMenuListItem 
                    linkTo="/spam"
                    primaryContent={STRINGS.SpamMenu}
                />
                <DrawerMenuListItem 
                    linkTo="/trash"
                    primaryContent={STRINGS.DeletedMenu}
                />
            </ul>
		);
    }

}

function mapStateToProps(store, ownProps) {
    return {
    };
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators({
    },
    dispatch);
}

DrawerMenuList.propTypes = propTypes;
DrawerMenuList.defaultProps = defaultProps;

export default connect(mapStateToProps, mapDispatchToProps)(DrawerMenuList);