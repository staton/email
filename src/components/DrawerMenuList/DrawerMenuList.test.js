import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { DrawerMenuList } from './DrawerMenuList';

describe('<DrawerMenuList />', () => {

    it('renders correctly', () => {
        const wrapper = shallow(<DrawerMenuList />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

});