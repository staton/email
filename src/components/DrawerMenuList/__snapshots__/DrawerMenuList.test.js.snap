// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerMenuList /> renders correctly 1`] = `
<ul
  className="DrawerMenuList"
>
  <Connect(DrawerMenuListItem)
    badge={205}
    icon={<Unknown />}
    primaryContent="Inbox"
    secondaryContent="test"
  />
  <Connect(DrawerMenuListItem)
    primaryContent="Drafts"
  />
  <Connect(DrawerMenuListItem)
    primaryContent="Sent"
  />
  <Connect(DrawerMenuListItem)
    primaryContent="Spam"
  />
  <Connect(DrawerMenuListItem)
    primaryContent="Deleted"
  />
</ul>
`;
